### ISSUE TITLE
Add Filter to Search

### ISSUE THREAD

**Issue by ramimac** [ryanhugh/searchneu#1]
Right now there is no easy way to narrow down generic search terms to only show results that are Employees, or Classes.

**Comment by ryanhugh** [comment]
Yup! Filters are definitely on the to do list. Also want to add Buildings at NEU, Student clubs, some other data sources and a bunch of other things.

**Comment by ryanhugh** [comment]
Things that we can add support for filtering over:
- Employees vs Classes
- Online vs On-Ground
- Class attributes (NU Path/NU Core)
- Subjects
- Seats availible
- Time of day
- Has prerequisites vs does not have prerequisites
- etc

The most-requested filter is definitely filtering over class attributes. Lets add filters after the class attributes data is scraped.

**Comment by edward-shen** [comment]
Bumping this, since I just had a friend suggest filtering as a new feature request.

**Comment by edward-shen** [comment]
How would filtering work? Is this something we can implement in our backend, e.g. in /common/search.js, or would we need to do client-side filtering?

**Comment by ryanhugh** [comment]
I'm thinking it would be in the backend and implemented pretty similar to searching only in a specific term work right now. The client would click a bunch of dropdowns, the selections would go on the state, be sent to the server as part of the request, and then `common/search.js` would perform the search and return the results. The two caches (in the frontend and the backend) would have to be updated with the new results too. 

I miiight have some time to spend on this, not sure

**Comment by ryanhugh** [comment]
Also for keeping track of all of these in the URL we could do the same thing as Airbnb and just use query params. I'm thinking that the termId could be a part of the path because that is required, but the rest are just going to be optional. Eg:

```
https://searchneu.com/201830/cs?onlineOnly=true&nupath=writingintensive&seatsAvailible=true ...
```
